docs/source/configuration/advanced_configuration.md

```diff
@@ -262,58 +262,73 @@ Since both of the file names (`catalog.yml` and `catalog_globals.yml`) match the
-        conf_source: str,
-        env: str = None,
-        runtime_params: Dict[str, Any] = None,
-    ):
-        super().__init__(
-            conf_source=conf_source, env=env, runtime_params=runtime_params
-        )
-
-        # Register a customer resolver that adds up numbers.
-        self.register_custom_resolver("add", lambda *numbers: sum(numbers))
-
-    @staticmethod
-    def register_custom_resolver(name, function):
-        """
-        Helper method that checks if the resolver has already been registered and registers the
-        resolver if it's new. The check is needed, because omegaconf will throw an error
-        if a resolver with the same name is registered twice.
-        Alternatively, you can call `register_new_resolver()` with  `replace=True`.
-        """
-        if not OmegaConf.has_resolver(name):
-            OmegaConf.register_new_resolver(name, function)
-```
 
-In order to use this custom configuration loader, you will need to set it as the project configuration loader in `src/<package_name>/settings.py`:
+CONFIG_LOADER_CLASS = OmegaConfigLoader
 
-```python
-from package_name.custom_configloader import CustomOmegaConfigLoader
 
-CONFIG_LOADER_CLASS = CustomOmegaConfigLoader
-```
+def date_today():
+    return date.today()
 
-You can then use the custom "add" resolver in your `parameters.yml` as follows:
 
+CONFIG_LOADER_ARGS = {
+    "custom_resolvers": {
+        "add": lambda *my_list: sum(my_list),
+        "polars": lambda x: getattr(pl, x),
+        "today": lambda: date_today(),
+    }
+}
+```
+These custom resolvers are then registered using `OmegaConf.register_new_resolver()` under the hood and can be used in any of the
+configuration files in your project. For example, you can use the `add` or the `today` resolver defined above in your `parameters.yml` like this:
 ```yaml
 model_options:
-  test_size: ${add:1,2,3}
+  test_size: "${add:1,2,3}"
   random_state: 3
+
+date: "{today:}"
 ```
+The values of these parameters will be computed at access time and will be passed on to your nodes.
+Resolvers can also be used in your `catalog.yml`. In the example below, we use the `polars` resolver defined above to pass non-primitive
+types to the catalog entry.
 
+```yaml
+my_polars_dataset:
+  type: polars.CSVDataSet
+  filepath: data/01_raw/my_dataset.csv
+  load_args:
+    dtypes:
+      product_age: "${polars:Float64}"
+      group_identifier: "${polars:Utf8}"
+    try_parse_dates: true
+```
+`OmegaConf` also comes with some [built-in resolvers](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html#built-in-resolvers)
+that you can use with the `OmegaConfigLoader` in Kedro. All built-in resolvers except for [`oc.env`](https://omegaconf.readthedocs.io/en/latest/custom_resolvers.html#oc-env)
+are enabled by default. `oc.env` is only turned on for loading credentials. You can, however, turn this on for all configurations
+through your project's `src/<package_name>/settings.py` in a similar way:
```

Comment:
I think we should add here that we don't necessarily recommend enabling `oc.env` everywhere. Maybe mark this as "advanced feature" ? I still very much stand by the comment I made here about allowing `oc.env` everywhere: https://github.com/kedro-org/kedro/issues/2407#issuecomment-1463610199

Reply:
Do you think I should remove this part entirely or just add a disclaimer of sorts saying we don't recommend it but you can still do it?

Reply:
I think we should probably keep it because we have had questions about it from users, but indeed add the disclaimer so beginner users will at least understand this isn't recommended unless you know what you're doing.